### ISSUE TITLE
Adding stub file of cm.py

### ISSUE THREAD

**Issue by beskep** [1313e/CMasher#110]
**Is your feature request related to a problem? Please describe.**

Adding stub file of `cm.py` would be very helpful for auto completing and type checking.
<br>

**Describe the solution you would like**

`cm.pyi` would be like the following:

```python
from matplotlib.colors import ListedColormap as LC

CMAP_DCT = dict[str, LC]
cmap_d: CMAP_DCT
cmap_cd: dict[str, CMAP_DCT]

amber: LC
amber_r: LC
amethyst: LC
amethyst_r: LC
apple: LC
apple_r: LC
arctic: LC
arctic_r: LC
⋮
```

I generated the stub file with this script:

```python
from pathlib import Path

import cmasher.cm as cmrcm
from cmasher._known_cmap_types import _CMASHER_BUILTIN_MAP_TYPES

if __name__ == "__main__":
    pyi = Path(cmrcm.__file__).with_suffix(".pyi")

    with pyi.open("w", encoding="utf-8") as f:
        f.write(
            """from matplotlib.colors import ListedColormap as LC

CMAP_DCT = dict[str, LC]
cmap_d: CMAP_DCT
cmap_cd: dict[str, CMAP_DCT]

"""
        )

        for cmap in _CMASHER_BUILTIN_MAP_TYPES:
            if cmap.startswith("cmr."):
                continue

            f.write(f"{cmap}: LC\n")
            f.write(f"{cmap}_r: LC\n")

```

<br>

**Additional context**

![image](https://github.com/1313e/CMasher/assets/26139969/4dadd178-0f2c-4a6d-af1a-1322728ad4c3)

**Comment by neutrinoceros** [comment]
This would be a useful addition indeed, PR welcome !
The script used for generation would need to be added to `scripts` too.

**Comment by beskep** [comment]
#111 Thank you for kind response. This is my first PR :)

**Comment by neutrinoceros** [comment]
closed in #112 , thanks again !
